app/javascript/packs/webauthn-authenticate.ts

```diff
@@ -41,8 +39,6 @@ function webauthn() {
       })
       .catch((error: Error) => {
         (document.getElementById('webauthn_error') as HTMLInputElement).value = error.name;
-        (document.getElementById('platform') as HTMLInputElement).value =
-          String(webauthnPlatformRequested);
```

Comment:
...but since we were still assigning a value here from the non-existent `data-` attribute, the form value was being overridden.

The solution is to just not set the value when the error happens, so that the form value remains undisturbed.